library/core/src/ptr/const_ptr.rs

```diff
@@ -101,6 +100,35 @@ impl<T: ?Sized> *const T {
         if self.is_null() { None } else { unsafe { Some(&*self) } }
     }
 
+    /// Returns `None` if the pointer is null, or else returns a reference to
+    /// the value wrapped in `Some`.
+    ///
+    /// While this method is useful for null-safety, it is important to note
+    /// that this is still an unsafe operation because the returned value could
+    /// be pointing to invalid memory.
```

Comment:
"Invalid memory" in the sense of "containing invalid data" is not a concern.

But the pointer could be a non-null dangling pointer. That would be a problem.